### PR TITLE
Merge bdb and dc45

### PR DESCRIPTION
**Pull request by jsjiang**
@sfisher @rushirajnenuji Hi Scott and Rushiraj,
I created a new branch and pull request for the DataCite 4.5 upgrade as the old branch does not work on the new AL2023 server. 

We can use this pull request to review the changes required for DataCite Schema 4.5 upgrade. The requirements are listed in ticket https://github.com/CDLUC3/ezid/issues/511. The procedure is documented here: https://github.com/CDLUC3/ezid-docs-internal/blob/main/docs/how_to_upgrade_datacite_schema.md.

This branch is on ezid-dev so you can test:

Create DataCite DOI use the advanced create form on UI
Create DataCite DOI use API
I will need to find ways to test backward supporting of older versions:

2.1 and 2.2: supported currently. Probably still need to be supported
3 (3.0 and 3.1): added support this time with the 4.5 upgrade
4.x (x<5): I think there is no need to make special changes for support older subversion of 4
Please review. We can discuss change details at our dev-sync tomorrow.

Thank you

Jing

### CODE REVIEW

File: xsd/datacite-kernel-4/metadata.xsd [hunk 1]
```diff
@@ -524,7 +528,7 @@ Use the complete title of a license and include version information if applicabl
                     </xs:element>
                     <xs:element name="number" minOccurs="0">
                       <xs:annotation>
-                        <xs:documentation>Issue number or name of the related item.</xs:documentation>
+                        <xs:documentation>Number of the related item e.g. report number of article number.</xs:documentation>
```
Comment (sfisher) [inline xsd/datacite-kernel-4/metadata.xsd:531]: I think maybe this should be `e.g. report number or article number` (`or` instead of `of`).

**Review by sfisher — APPROVED** [review]
This looks good to me, @jsjiang .  I ran the branch on my machine and was able to add items for the `publisher`.  It also notified me if I didn't fill in the required field in advanced.

File: xsd/datacite-kernel-4/metadata.xsd (hunk 1 above)
Comment (jsjiang) [reply xsd/datacite-kernel-4/metadata.xsd:531]: This file was downloaded from DataCite. We probably want to keep the statement as is.

**Comment by sfisher** [comment]
thanks for the update to the fields, Jing.  I think those extra changes look good.  Where are all the abbreviations for the resource types (like DataPaper, Dataset) coming from?  Are those internal abbreviations in EZID?

**Comment by jsjiang** [comment]
> thanks for the update to the fields, Jing. I think those extra changes look good. Where are all the abbreviations for the resource types (like DataPaper, Dataset) coming from? Are those internal abbreviations in EZID?

@sfisher Hi Scott, the resource types come from DataCite schema. The two digits code are defined by EZID for sorting purpose. However, there were coding mistakes in the past. Also, I don't really understand how these codes are used. I need to look into these codes closely.

Jing

**Pull request merged**